### PR TITLE
Add gate labels

### PR DESCRIPTION
**Pull request by vodovozovaliza**
Related to issue #850. This PR allows to change `name` parameter of `qibo.gates.Gate` and 2 other labels are added:
- `name`: label of the gate. Can be changed by a user.
- `draw_label`: a symbol for drawing a circuit. If `None` or empty, `gate.name[:4]` is used.
    This parameter allows to remove `qibo.gates.DRAW_LABELS` dictionary.
- `@property qasm_label`: for gates supported by OpenQASM.

For example:
```python
from math import pi 
from qibo import gates
from qibo.models import Circuit

rx1 = gates.RX(0, pi/2)
rx1.name = "RX90"
rx1.draw_label = "RX1"
rx2 = gates.RX(1, 3 * pi/2)
rx2.name = "RX270"
rx2. draw_label = "RX2"

circuit = Circuit(2)
circuit.add(rx1)
circuit.add(rx2)
print(circuit.draw())
```
prints
```python
q0: ─RX1─
q1: ─RX2─
```
instead of 
```python
q0: ─RX─
q1: ─RX─
```
As mentioned in #850, the `name` can be used for NoiseModels or for controlling parameters of the quantum hardware.

Checklist:
- [ ] Reviewers confirm new code works as expected.
- [x] Tests are passing.
- [x] Coverage does not decrease.
- [x] Documentation is updated.

### CODE REVIEW

File: src/qibo/models/circuit.py [hunk 1]
```diff
@@ -1272,12 +1272,15 @@ def read_args(args):
     def _update_draw_matrix(self, matrix, idx, gate, gate_symbol=None):
         """Helper method for :meth:`qibo.models.circuit.Circuit.draw`."""
         if gate_symbol is None:
-            if gate.name not in gates.DRAW_LABELS:  # pragma: no cover
+            if gate.label is not None and gate.label != gate.name:
+                gate_symbol = gate.label[:4]
+            elif gate.name in gates.DRAW_LABELS:  # pragma: no cover
+                gate_symbol = gates.DRAW_LABELS.get(gate.name)
+            else:
                 raise_error(
```
Comment (renatomello) [inline src/qibo/models/circuit.py:1280]: a test is missing for this line

**Comment by renatomello** [comment]
the parameters `label` and `name` seem to be somewhat redundant. Maybe there's a way to eliminate the redundancy by using class type instead of `name`, and then using `name` as `label` is being intended. 

For instance, any time there's a `if self.name  == "<insert-name>"` appears, we could use `if isinstance(self, "insert class type") is True`. Then, `self.name` is free to be used as a label.

I haven't looked at every instance in which `self.name` is being used, so I don't know if this suggestion even works.

**Comment by stavros11** [comment]
> the parameters `label` and `name` seem to be somewhat redundant. Maybe there's a way to eliminate the redundancy by using class type instead of `name`, and then using `name` as `label` is being intended.

Looking at the changes, I also agree that we could keep only one attribute for `label` and `name` (feel free to choose whichever you prefer).

> For instance, any time that `if self.name == "<insert-name>"` appears, we could use `if isinstance(self, "insert class type") is True`. Then, `self.name` is free to be used as a label.
> 
> I haven't looked at every instance in which `self.name` is used, so I don't know if this suggestion even works and/or would break other things.

I also do not recall all the places where `name` is used, but most likely should be possible to replace it with `isinstance`. @vodovozovaliza if you also agree with us, try to have a look if you can make this change and use the `name` as `label`. I think you can easily find all lines that `name` appears in the code using some `grep` command in linux.

**Comment by vodovozovaliza** [comment]
@renatomello @stavros11 thank you for the comments, I will look into replacing `name` and `label` with just one of them.

**Comment by vodovozovaliza** [comment]
Thank you @stavros11. Turning `qasm_label` into `@property` is a very good idea.

**Comment by renatomello** [comment]
is this dictionary https://github.com/qiboteam/qibo/blob/e6afcd583a21b63b5a52b5a105d0e1074a764c7a/src/qibo/gates/gates.py#L8 still needed?

**Comment by vodovozovaliza** [comment]
> is this dictionary
> 
> https://github.com/qiboteam/qibo/blob/e6afcd583a21b63b5a52b5a105d0e1074a764c7a/src/qibo/gates/gates.py#L8
> 
> still needed?

Yes, this dictionary is needed for `Circuit.from_qasm` method

**Review by renatomello — APPROVED** [review]

**Comment by renatomello** [comment]
@stavros11 could you also check this before we merge?

**Comment by renatomello** [comment]
@vodovozovaliza since `name` became the de facto `label`, could you rename this PR?

**Review by stavros11 — APPROVED** [review]
Thanks for the updates, looks good to me too. I just left a comment regarding removing the `PARAMETRIZED_GATES` set. This is not strictly required to merge but since you already cleaned the other dictionaries in gates.py, it may be a good idea to do so.

If understand correctly now the user is able to control both the `name` and the `draw_label` seperately. Is this okay for your applications or you prefer the `draw_label` to change every time the `name` changes? If that's the case we may need to have a setter or drop completely the `draw_label` and use only the `name` (but then we lose the flexibility of adjusting them seperately).

**Pull request merged**